### PR TITLE
HW 1

### PR DESCRIPTION
**Pull request by AntonKalenskiy**

### CODE REVIEW

**Comment by AntonKalenskiy** [comment]
Sorry, I didn`t know how to separate these tasks...

File: src/test/resources/application.properties [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#DB configuration
+
+spring.datasource.url=jdbc:h2:mem:testdb
+spring.datasource.driverClassName=org.h2.Driver
+spring.datasource.username=sa
+spring.datasource.password=password
+spring.jpa.database-platform=org.hibernate.dialect.H2Dialect
+spring.h2.console.enabled=true
+
+
```
Comment (okuzan) [inline src/test/resources/application.properties:10]:
```suggestion
```

File: src/main/java/com/springframework/boot/onlinebookstore/controller/DemoController.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.springframework.boot.onlinebookstore.controller;
+
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class DemoController {
+
```
Comment (okuzan) [inline src/main/java/com/springframework/boot/onlinebookstore/controller/DemoController.java:8]:
```suggestion
```

File: src/main/java/com/springframework/boot/onlinebookstore/dto/mapper/BookMapper.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import com.springframework.boot.onlinebookstore.dto.CreateBookRequestDto;
+import com.springframework.boot.onlinebookstore.model.Book;
+import org.mapstruct.Mapper;
+
+@Mapper(config = MapperConfig.class)
+public interface BookMapper {
+    BookDto toDto(Book book);
+
+    Book toModel(CreateBookRequestDto requestDto);
+
```
Comment (okuzan) [inline src/main/java/com/springframework/boot/onlinebookstore/dto/mapper/BookMapper.java:14]:
```suggestion
```

File: src/main/java/com/springframework/boot/onlinebookstore/config/MapperConfig.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+import org.mapstruct.InjectionStrategy;
+import org.mapstruct.NullValueCheckStrategy;
+
+@org.mapstruct.MapperConfig(componentModel = "spring",
+                            injectionStrategy = InjectionStrategy.CONSTRUCTOR,
+                            nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS,
+                            implementationPackage = "<PACKAGE_NAME>.impl")
+public class MapperConfig {
+
```
Comment (okuzan) [inline src/main/java/com/springframework/boot/onlinebookstore/config/MapperConfig.java:11]:
```suggestion
```

**Review by okuzan — APPROVED** [review]

**Pull request merged**